presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -104,7 +104,8 @@ public synchronized Response install(@ApiParam("Url to fetch package") String lo
             @ApiResponse(code = 409, message = "Presto is running")
     })
     public synchronized Response upgrade(@ApiParam("Url to fetch package") String locationToFetchPackage,
-            @QueryParam("checkDependencies") @DefaultValue("true") @ApiParam("If false, disables dependency checking") boolean checkDependencies)
+            @QueryParam("checkDependencies") @DefaultValue("true") @ApiParam("If false, disables dependency checking") boolean checkDependencies,
+            @QueryParam("preserveConfigs") @DefaultValue("false") @ApiParam("If true, preserves all the config files") boolean preserveConfigs)
```

Comment:
Maybe call this "keepConfig"? It's a bit more succinct, and has the same meaning.